### PR TITLE
Testcase add for projectEnrich

### LINKED ISSUE #1441: JKube Kit : Add unit test for ProjectEnricher
### Description

[ProjectEnricher](https://github.com/eclipse/jkube/blob/master/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricher.java) doesn't seem to be covered by unit tests

- [x] Add a unit test to verify Project is generated as expected

### PR DESCRIPTION
**Pull request by Devamparikh**
## Description
Fix #1441

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have implemented unit tests to cover my changes
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I tested my code in Kubernetes
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: quickstarts/maven/spring-boot/pom.xml [hunk 1]
```diff
@@ -146,7 +146,7 @@
           <plugin>
             <groupId>org.eclipse.jkube</groupId>
             <artifactId>openshift-maven-plugin</artifactId>
-            <version>${jkube.version}</version>
```
Comment (rohanKanojia) [inline quickstarts/maven/spring-boot/pom.xml:149]: Why is this change required?

Comment (rohanKanojia) [inline quickstarts/maven/spring-boot/pom.xml:149]: Why is this change required?

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package org.eclipse.jkube.enricher.generic.openshift;
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:1]: Add license headers to new file by `mvn -N license:format`

Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:1]: Add license headers to new file by `mvn -N license:format`

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import org.eclipse.jkube.kit.config.resource.PlatformMode;
+import org.eclipse.jkube.kit.config.resource.ResourceConfig;
+import org.eclipse.jkube.kit.enricher.api.EnricherContext;
+import org.eclipse.jkube.kit.enricher.api.JKubeEnricherContext;
+import org.junit.Test;
+
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.*;
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:17]: We don't use wildcard imports in this repository

Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:17]: We don't use wildcard imports in this repository

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                        .artifactId("artifact-id")
+                        .build())
+                .build();
+    }
+
+    @Test
+    public void convertNamespaceResourceToProject() {
+        // Given
+        Properties properties = new Properties();
+        properties.put("jkube.enricher.jkube-namespace.namespace", "example");
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:39]: This property shouldn't be required.

Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:39]: This property shouldn't be required.

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java (hunk 3 above)
Comment (rohanKanojia) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:17]: On second look, This import shouldn't be required.

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.*;
+
+public class ProjectEnricherTest {
+
+    private EnricherContext context;
+
+    public void setExpectations(Properties properties, ResourceConfig resourceConfig) {
+        context = JKubeEnricherContext.builder()
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:24]: You can mock JKubeEnricherContext class instead:
```java
context = mock(JKubeEnricherContext.class, RETURNS_DEEP_STUBS);
```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    public void convertNamespaceResourceToProject() {
+        // Given
+        Properties properties = new Properties();
+        properties.put("jkube.enricher.jkube-namespace.namespace", "example");
+        setExpectations(properties, new ResourceConfig());
+        final KubernetesListBuilder klb = new KubernetesListBuilder();
+        // When
+        new ProjectEnricher((JKubeEnricherContext) context).create(PlatformMode.openshift, klb);
+        // Then
+        assertThat(klb.build().getItems()).isEmpty();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:45]: This test doesn't seem to be verifying what it's named. You need to test this scenario:

1. KubernetesListBuilder contains a Namespace object
2. When ProjectEnricher.create() is invoked
3. KubernetesListBuilder should contain Project object

Translating this to code should look like this:
```java
        final KubernetesListBuilder klb = new KubernetesListBuilder();
        klb.addToItems(new NamespaceBuilder().withNewMetadata().withName("foo").endMetadata());
        // When
        new ProjectEnricher((JKubeEnricherContext) context).create(PlatformMode.openshift, klb);
        // Then
        assertThat(klb.build())
            .extracting(KubernetesList::getItems)
            .asList()
            .hasSize(1)
            .first()
            .isInstanceOf(Project.class)
            .hasFieldOrPropertyWithValue("metadata.name", "foo");
```

Comment (Devamparikh) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java]: I think we should also add given KubernetesListBuilder is empty then no project will be created. What do you think, will that help in increasing coverage for ProjectEnricher?

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.enricher.generic.openshift;
+
+import io.fabric8.kubernetes.api.model.*;
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:16]: Please remove this wildcard import

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+public class ProjectEnricherTest {
+
+    private EnricherContext context;
+
+    public void setExpectations(Properties properties, ResourceConfig resourceConfig) {
+        context = mock(JKubeEnricherContext.class, RETURNS_DEEP_STUBS);
+    }
+
+    @Test
+    public void convertNamespaceResourceToProject() {
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java:39]:
```suggestion
    public void create_whenKubernetesListHasNamespace_thenNamespaceConvertedToProject() {
```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java (hunk 6 above)
Comment (rohanKanojia) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java]: Sounds good to me.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
